segmentation/include/pcl/segmentation/impl/extract_clusters.hpp

```diff
@@ -86,30 +87,39 @@ pcl::extractEuclideanClusters (const PointCloud<PointT> &cloud,
         if (nn_indices[j] == -1 || processed[nn_indices[j]])        // Has this point been processed before ?
```

Comment:
You're not allowed to rely on index being -1. Does `radiusSearch` provide overloads which do not rely on this behavior?

Reply:
I am not sure if we even need `nn_indices[j] == -1`. I did some tracing  and couldn't find setting of `nn_indices[j]` to -1. The commit history also doesn't give insight on why it was abruptly added in between since it seems to link to a PR before usage of GitHub.